Update documentation (#305)
carenalgas/popochiu: docs/src/how-to-develop-a-game/changing-rooms.md

---
weight: 4025
---

## How to change rooms

You can change a room by using ```E.goto_room('RoomName')``` or ```R.current_room = R.RoomName```. The room names can be easily found in the Main tab of the Popochiu Dock under the rooms section.

In this example I have a hotspot called 'RoomTwoExit' and switching to a room called 'RoomTwo' in real life you would use more meaningful and descriptive names:
```

> func _on_click() -> void:
>     await C.player.walk_to_clicked()
>     await C.player.face_down()
>     E.goto_room('RoomTwo')
> 
> # Allows the player to double click hot spot to change rooms without walking to the hotspot
> func _on_double_click() -> void:
>     await C.player.teleport_to_hotspot("RoomTwoExit")
> 
>     # Calling _on_click will run all the code in the _on_click() function so no code is duplicated
>     _on_click()

Reviewer: I understand this example but I would rewrite it in a way that does not imply functions cross-calling. Beginners may think this is a canonical way to achieve the result.

I propose this version:

```suggestion
func _on_click() -> void:
    await C.player.walk_to_clicked()
    await C.player.face_down()
    E.goto_room('RoomTwo')

# Allows the player to double click hot spot to change rooms without walking to the hotspot
func _on_double_click() -> void:
    # Just change room
    E.goto_room('RoomTwo')
```
